src/api/ApiClient.js

```diff
@@ -46,6 +46,20 @@ export default class ApiClient {
     });
   }
 
+  static refreshTokenPromise = null;
+
+  refreshToken() {
+    ApiClient.refreshTokenPromise = this.post('auth/refresh', {
+      refreshToken: window.localStorage.getItem('refreshToken'),
+    }).then((data) => {
+      if (data.code === 401) {
+        Promise.reject();
```

Comment:
WTF? 
1. Promise should be rejected with error object!
2. You don't return it anywhere - GC will collect it almost immediately...